CorsixTH/Lua/game_ui.lua

```diff
@@ -643,10 +643,82 @@ function GameUI:onMouseUp(code, x, y)
     else -- No room chosen yet, but about to edit one.
       if button == "left" then -- Take the clicked one.
         local room = self.app.world:getRoom(self:ScreenToWorld(x, y))
-        if room and not room.crashed then
-          self:setCursor(self.waiting_cursor)
-          self.edit_room = room
-          room:tryToEdit()
+        if room then
+          if not room.crashed then
+            self:setCursor(self.waiting_cursor)
+            self.edit_room = room
+            room:tryToEdit()
+          else
+            if self.app.config.remove_destroyed_rooms then
```

Comment:
Fold this new code into a function, probably even more than one. Having to find a matching 'end' 80 lines down is a mess!

Reply:
Indeed, it was just a slightly massaged guts of UIEditRoom:returnToDoorPhase, so can probably massage another method there to cover both scenarios.